src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/ModelMetadataUtilities.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Web.CodeGeneration;
+using Microsoft.VisualStudio.Web.CodeGeneration.EntityFrameworkCore;
+
+namespace Microsoft.VisualStudio.Web.CodeGenerators.Mvc
+{
+    internal static class ModelMetadataUtilities
```

Comment:
>ModelMetadataUtilities [](start = 26, length = 22)

Even though its ok, it feels a bit weird to have too many utility methods like this.. in general i would stay away from "utilities" unless they are really atomic and specific to some particular type that we extend... This kind of code looks like it could spread in between participating objects , however in this particular case its unclear how to do it without redesigning ValidationUtility and other classes. So its ok , just a thought ...